core/src/main/scala/org/apache/spark/sql/delta/ActiveOptimisticTransactionRule.scala

```diff
@@ -50,28 +99,58 @@ class ActiveOptimisticTransactionRule(spark: SparkSession) extends Rule[LogicalP
 
     def transform(plan: LogicalPlan): LogicalPlan =
       transformSubqueries(plan) transform {
-        case p @ PhysicalOperation(_, filters, DeltaTable(fileIndex: TahoeLogFileIndex))
-            if !fileIndex.isTimeTravelQuery && OptimisticTransaction.getActive().nonEmpty =>
```

Comment:
This check seems to have gotten lost, so everything is now turning into a PinnedTahoeFileIndex regardless of if there's an active transaction?

Reply:
Good catch. This should be resolved when moving the code to [PrepareDeltaScan](https://github.com/delta-io/delta/blob/503881673da016ba29b5ca79374819a199a2d613/core/src/main/scala/org/apache/spark/sql/delta/stats/PrepareDeltaScan.scala).